lib/node_modules/@stdlib/utils/error-to-json/test/test.js

```diff
@@ -0,0 +1,456 @@
+		values[ i ].errno = expected[ i ];
+		json = toJSON( values[ i ] );
+		t.equal( json.errno, expected[ i ], 'errno equal to \'' + expected[ i ] + '\'' );
+	}
+	t.end();
+});
+
+test( 'if a `syscall` property is present, the JSON object includes a failed system call string', function test( t ) {
+	var expected;
+	var values;
+	var json;
+	var i;
+
+	values = [
+		new Error(),
+		new EvalError(),
+		new TypeError(),
+		new ReferenceError(),
+		new SyntaxError(),
+		new URIError(),
+		new RangeError()
+	];
+
+	expected = [
+		'a',
+		'b',
+		'c',
+		'd',
+		'e',
+		'f',
+		'g'
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		values[ i ].syscall = expected[ i ];
+		json = toJSON( values[ i ] );
+		t.equal( json.syscall, expected[ i ], 'syscall equal to \'' + expected[ i ] + '\'' );
+	}
+	t.end();
+});
+
+test( 'if a provided error has additional enumerable properties, the function will include these properties and their values in the output JSON', function test( t ) {
+	var json;
+	var err;
+
+	// Data descriptor...
+	err = new Error( 'errrr' );
+	err.beep = 'boop';
+	err.boop = 'beep';
+
+	json = toJSON( err );
+	t.equal( json.beep, err.beep, 'data descriptor' );
+	t.equal( json.boop, err.boop, 'data descriptor' );
+
+	// Accessor descriptor...
+	err = new Error( 'errrr' );
+	Object.defineProperty( err, 'beep', {
+		'enumerable': true,
+		'configurable': true,
+		'get': function get() {
+			return 'boop';
+		}
+	});
+	Object.defineProperty( err, 'boop', {
+		'enumerable': true,
+		'configurable': false,
+		'get': function get() {
+			return 'beep';
+		}
+	});
+
+	json = toJSON( err );
+	t.equal( json.beep, err.beep, 'accessor descriptor' );
+	t.equal( json.boop, err.boop, 'accessor descriptor' );
+
+	t.end();
+});
+
+test( 'the function deep copies enumerable properties', function test( t ) {
+	var json;
+	var err;
+
+	// Deep equal...
+	err = new Error( 'errrr' );
+	err.arr = [ 1, 2, [ 3, 4, 5 ] ];
+
+	json = toJSON( err );
+	t.notEqual( json.arr, err.arr, 'new instances' );
+	t.deepEqual( json.arr, err.arr, 'deep equal' );
+
+	t.end();
+});
+
+test( 'custom errors (proto)', function test( t ) {
+	var CustomError;
+	var types;
+	var ctors;
+	var json;
+	var err;
+	var i;
+
+	ctors = [
+		Error,
+		TypeError,
+		SyntaxError,
+		ReferenceError,
+		URIError,
+		EvalError,
+		RangeError
+	];
+
+	types = [
+		'Error',
+		'TypeError',
+		'SyntaxError',
+		'ReferenceError',
+		'URIError',
+		'EvalError',
+		'RangeError'
+	];
+
+	for ( i = 0; i < ctors.length; i++ ) {
+		CustomError = createClass1( ctors[ i ] );
+		err = new CustomError( 'custom error' );
+		json = toJSON( err );
+
+		t.equal( json.type, types[ i ], 'type equal to ' + types[ i ] );
+		t.equal( json.message, err.message, 'equal messages' );
+		t.equal( json.name, err.name, 'equal names' );
+		t.equal( json.stack, err.stack, 'equal stack trace' );
+	}
+	t.end();
+});
+
+if ( hasClassSupport ) {
+	test( 'custom errors (subclass; ES2015)', function test( t ) {
```

Comment:
Use `opts.skip`. Can set `opts.skip` at the top of file like we do for other packages.